editor/editor_dock_manager.cpp

```diff
@@ -74,10 +75,13 @@ void DockSplitContainer::add_child_notify(Node *p_child) {
 	SplitContainer::add_child_notify(p_child);
 
 	Control *child_control = nullptr;
-	for (int i = 0; i < 2; i++) {
-		Control *split = get_containable_child(i);
-		if (p_child == split) {
-			child_control = split;
+	for (int i = 0; i < get_child_count(false); i++) {
+		Control *c = Object::cast_to<Control>(get_child(i, false));
+		if (!c || c->is_set_as_top_level()) {
```

Comment:
Not checking for `is_visible()`?

Reply:
It's intentional, We want to connect / disconnect dock slots to `visibility_changed`, even if they aren't currently visible. Especially when disconnecting, since the children's visiblility changes a lot. 
The reason `_update_visibility()` checks `is_visible()` is because it is checking specifically if any children are visible.